pkg/dependency/parser/php/composer/parse.go

```diff
@@ -44,7 +44,7 @@ func (p *Parser) Parse(r xio.ReadSeekerAt) ([]ftypes.Package, []ftypes.Dependenc
 	if err != nil {
 		return nil, nil, xerrors.Errorf("read error: %w", err)
 	}
-	if err = jfather.Unmarshal(input, &lockFile); err != nil {
+	if err = json.Unmarshal(input, &lockFile, json.WithUnmarshalers(xjson.UnmarshalerWithObjectLocation(input))); err != nil {
```

Comment:
All JSON parsing requires `xjson.UnmarshalerWithObjectLocation` now and it doesn't look straightforward. What if defining `Unmarshal` in xjson?

```
func Unmarshal(data []byte, v any) error {
	return json.Unmarshal(data, v, json.WithUnmarshalers(xjson.UnmarshalerWithObjectLocation(data)))
}
```

Then, we can easily write json unmarshaling here.

```suggestion
	if err = xjson.Unmarshal(input, &lockFile); err != nil {
```

Reply:
it make sense.
Updated in https://github.com/aquasecurity/trivy/pull/8591/commits/c5e9dc20514377366a08e5f7b4dcea682969e35c